src/lib/vault/getFormSubmission.ts

```diff
@@ -0,0 +1,40 @@
+import { GetCommand } from "@aws-sdk/lib-dynamodb";
+import { dynamodbClient } from "@lib/awsServicesConnector";
+
+interface FormSubmission {
+  content: string;
+}
+
+export async function getFormSubmission(
+  formId: string,
+  submissionName: string,
+): Promise<FormSubmission | undefined> {
+  try {
+    const response = await dynamodbClient.send(
+      new GetCommand({
+        // biome-ignore lint/style/useNamingConvention: <explanation>
+        TableName: "Vault",
+        // biome-ignore lint/style/useNamingConvention: <explanation>
+        Key: { FormID: formId, NAME_OR_CONF: `NAME#${submissionName}` },
+        // biome-ignore lint/style/useNamingConvention: <explanation>
+        ProjectionExpression: "FormSubmission",
+      }),
+    );
+
+    if (response.Item === undefined) {
+      return undefined;
```

Comment:
Should we be a bit more explicit? like returning "invalid form id or submission name" ?

Reply:
I wouldn't.  At the route level it's caught and a 404 is returned to the client.

Reply:
Yeah and even if the format of both `formId` and `submissionName` is valid we still would not be able to know which does not exist because DynamoDB does not tell us and just returns an undefined property.
Also, in the same spirit of not showing whether the username or password is wrong in a login page, I think it would make sense not to tell any possible attacker that he is getting closer to finding an existing form submission.